Source/Fuse.Scripting.JavaScript/Tests/Model/Model.Test.uno

```diff
@@ -0,0 +1,581 @@
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				Assert.AreEqual("a", e.a.UseValue);
+				Assert.AreEqual("b", e.b.UseValue);
+
+				e.callStep1.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual("c", e.a.UseValue);
+				Assert.AreEqual("b", e.b.UseValue);
+
+				e.callStep2.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual("c", e.a.UseValue);
+				Assert.AreEqual("d", e.b.UseValue);
+
+				e.callStep3.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual("c", e.a.UseValue);
+				Assert.AreEqual("e", e.b.UseValue);
+			}
+		}
+
+		[Test]
+		public void Disconnected()
+		{
+			var e = new UX.Model.Disconnected();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				Assert.AreEqual(5,e.a.Value);
+
+				e.callUpdateNext.Perform();
+				e.callSwap.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual(11,e.a.Value);
+
+				e.callUpdateNext.Perform();
+				e.callSwap.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual(6,e.a.Value);
+			}
+		}
+
+		[Test]
+		public void AltEntry()
+		{
+			var e = new UX.Model.AltEntry();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				Assert.AreEqual( "a", e.a.StringValue );
+
+				e.callSetB.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual( "b", e.a.StringValue );
+			}
+		}
+
+		[Test]
+		public void MultiBasic()
+		{
+			var e = new UX.Model.Multi();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				Assert.AreEqual( 1, e.a.Value );
+				Assert.AreEqual( 2, e.b.Value );
+				Assert.AreEqual( 3, e.c.Value );
+			}
+		}
+
+		[Test]
+		public void Bind()
+		{
+			var e = new UX.Model.Bind();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				Assert.AreEqual( "bop", e.u.v.StringValue );
+				Assert.AreEqual( 0, e.u.id.Value );
+				Assert.AreEqual( 11, e.u.Load );
+				Assert.AreEqual( 10, e.u.DefaultFromJS );
+
+				e.u.Value = "loppy";
+				root.StepFrameJS();
+				Assert.AreEqual( "loppy", e.u.v.StringValue );
+				Assert.AreEqual( 0, e.u.id.Value );
+
+				e.u.callIncrLoad.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual( 12, e.u.Load );
+			}
+		}
+
+		[Test]
+		public void Accessor()
+		{
+			var e = new UX.Model.Accessor();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				Assert.AreEqual( 110, e.v.Value );
+				Assert.AreEqual( 10, e.r.Value );
+				Assert.AreEqual( 220, e.d.Value );
+
+				e.v.Value = 115;
+				root.StepFrameJS();
+				Assert.AreEqual( 115, e.v.Value );
+				Assert.AreEqual( 15, e.r.Value );
+				Assert.AreEqual( 230, e.d.Value );
+
+				e.callIncr.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual( 116, e.v.Value );
+				Assert.AreEqual( 16, e.r.Value );
+				Assert.AreEqual( 232, e.d.Value );
+			}
+		}
+
+		[Test]
+		//TODO: working on this this one still (found the Each one)
```

Comment:
Sounds like this TODO should be finished up...

Reply:
I think the TODO just needs to be removed now.